template.tpl

```diff
@@ -407,10 +449,16 @@ var eventMetadata = {
   value: data.transactionValue,
 };
 
-// If there is product information, add it to the eventMetadata. The simple table is
-// of the correct format - an array of objects.
-if (data.products && data.products.length) {
-  eventMetadata.products = data.products;
+// If there is product information, add it to the eventMetadata.
+if (data.productInputType == "entryManual" && data.productsRows && data.productsRows.length) {
+  // The simple table is of the correct format - an array of objects.
+  // We can assign it directly.
+  eventMetadata.products = data.productsRows;
+} else if (data.productInputType == "entryJSON" && data.productsJSON && data.productsJSON.length) {
+  // The text input should represent JSON. We need to parse it.
+  // If it fails, products simply won't be defined.
+  var products = JSON.parse(data.productsJSON);
```

Comment:
```
var products = [];
try {
  products = JSON.parse(data.productsJSON);
} catch (e) {
  console.error("Malformed products data.  Skipping ...");
}
```
Otherwise an exception will get thrown with malformed data.  The script will stop processing, and the pixel won't fire.

try/catch should be broadly compatible with browsers: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/try...catch#browser_compatibility